lib/vrl/stdlib/src/is_json.rs

```diff
@@ -53,11 +99,48 @@ impl Function for IsJson {
         mut arguments: ArgumentList,
     ) -> Compiled {
         let value = arguments.required("value");
-        Ok(Box::new(IsJsonFn { value }))
+        let variant = arguments.optional_enum("variant", &variants())?;
+
+        match variant {
+            Some(raw_variant) => {
+                let variant = raw_variant.try_bytes().expect("variant not bytes");
```

Comment:
I suspect you picked up this pattern from other function implementations, but we should bubble up errors, even if the compiler _currently_ prevents this panic from triggering.

```suggestion
                let variant = raw_variant.try_bytes()?;
```

Reply:
Also converted errors to `Box<dyn DiagnosticMessage>` to make it works 
```
.map_err(|e| Box::new(e) as Box<dyn DiagnosticMessage>)
```